server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -915,17 +934,22 @@ public void removeIndex(final Index index, final IndexRemovalReason reason, fina
 
             listener.beforeIndexRemoved(indexService, reason);
             logger.debug("{} closing index service (reason [{}][{}])", index, reason, extraInfo);
-            indexService.close(extraInfo, reason == IndexRemovalReason.DELETED);
-            logger.debug("{} closed... (reason [{}][{}])", index, reason, extraInfo);
-            final IndexSettings indexSettings = indexService.getIndexSettings();
-            listener.afterIndexRemoved(indexService.index(), indexSettings, reason);
-            if (reason == IndexRemovalReason.DELETED) {
-                // now we are done - try to wipe data on disk if possible
-                deleteIndexStore(extraInfo, indexService.index(), indexSettings);
-            }
-        } catch (Exception e) {
-            logger.warn(() -> format("failed to remove index %s ([%s][%s])", index, reason, extraInfo), e);
-        }
+            // ES-8334 passthru enclosing method is async
+            indexService.close(
+                extraInfo,
+                reason == IndexRemovalReason.DELETED,
+                EsExecutors.DIRECT_EXECUTOR_SERVICE,
```

Comment:
Should this use the `shardCloseExecutor` instead?

```suggestion
                shardCloseExecutor,
```

Reply:
Ah thanks yes I missed that one, see e81ea9015cc

Reply:
... which of course was the one which broke all the things 🤦